### PR TITLE
Redesign: Page attachment panel

### LINKED ISSUE #6464: Redesign: Update page attachment panel
## Feature Description
See [Figma](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=3483%3A322)

- [ ] Update panel order
- [ ] Update the URL input
- [ ] Update the link text input

## Additional Context
Screenshot 2021-02-22 at 18 03 49

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by embarks**
## Context
- update the designs in the Page Attachment panel

## Summary
- Use the new design-system input instead of the ExpandedTextInput
- Update page attachment panel order

## Relevant Technical Choices
- `LinkInput` was already implemented to use the design system but the ticket had a reference to `link text`. I'm not sure if that meant to change the `onBlur` to update the link so that it included `www` or some subdomain. I wrote this to change the validation to inject a default subdomain `www`:

```
const SUBDOMAIN_PATTERN = /^([a-z]+:\/{2})?([\w-]+\.[\w-]+\..*)$/;

/**
 * Prepends a subdomain (default www) to a URL that doesn't have one
 *
 * @param {string} url URL.
 * @param {string} [subdomain=www] default subdomain to prepend
 * @return {string} the url with the subdomain prepended to it
 */
export function withSubdomain(url, subdomain = 'www') {
  const hasSubdomain = SUBDOMAIN_PATTERN.test(url);
  if (hasSubdomain) {
    return url;
  }
  const [protocol] = url.match(PROTOCOL_PATTERN) ?? [];
  return protocol
    ? `${protocol}${subdomain}.${url.slice(protocol.length)}`
    : `${subdomain}.${url}`;
}
```
It has some weirdness if you type in something like `test` it will change it to `www.test` and then if you go in and blur again it will change it to `www.www.test` (because it's not passing the `hasSubdomain` test). The protocol was nice and simple because it just prepended it and there's a limited number of protocol strings to detect. the subdomain test is more complex. So I ultimately did not include it in the scope and left the functionality the same there.

- I did not include a `clear` button on this input--we don't have a visual reference for that, but I did update the panel to re-insert the default text if the value was empty.

## User-facing changes

https://user-images.githubusercontent.com/41136059/110009246-fd0a7300-7cd9-11eb-9312-3321aa8a2d83.mp4

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. open a new story
2. add a background element to the page and select the background element
3. Assure the page attachment panel is last, except for layers
4. Add a link and a description to the inputs in the page attachment panel
5. Assure that inputs in the page attachment panel are to spec according to the [Figma](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=3483%3A322) (Learn about page attachment should **not** be in the panel)

### UAT

- [x] UAT should use the same steps as above.

### Does this PR have a security-related impact?
no

### Does this PR change what data or activity we track or use?
no

### Does this PR have a legal-related impact?
no

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6464

### CODE REVIEW

File: assets/src/edit-story/components/panels/design/pageAttachment/pageAttachment.js [hunk 1]
```diff
@@ -136,18 +133,19 @@ function PageAttachmentPanel() {
...
-            onBlur={(atts = {}) => {
-              const { onClear } = atts;
-              if (onClear) {
+            onBlur={({ target }) => {
+              if (!target.value) {
                 updatePageAttachment({ ctaText: defaultCTA });
+                _setCtaText(defaultCTA);
               } else {
                 cancelCTAUpdate();
                 updatePageAttachment({
```
Comment (samwhale) [inline assets/src/edit-story/components/panels/design/pageAttachment/pageAttachment.js]: 💭 should we move this to its own event handler for better readability? Seems a little beefy for an inline function.

File: assets/src/edit-story/components/panels/design/pageAttachment/pageAttachment.js [hunk 2]
```diff
@@ -108,16 +108,13 @@ function PageAttachmentPanel() {
   );
 
   const isDefault = _ctaText === defaultCTA;
+  const showTextInput = Boolean(url) && !isInvalidUrl;
   return (
     <SimplePanel
       name="pageAttachment"
       title={__('Page Attachment', 'web-stories')}
     >
       <LinkInput
```
Comment (samwhale) [inline assets/src/edit-story/components/panels/design/pageAttachment/pageAttachment.js:137]: dope this already uses the input from the design system!

**Review by samwhale — APPROVED** [review]
amazzing

File: assets/src/edit-story/components/panels/design/pageAttachment/pageAttachment.js [hunk 3]
```diff
@@ -30,7 +30,7 @@ import { isValidUrl, withProtocol } from '../../../../utils/url';
 import useElementsWithLinks from '../../../../utils/useElementsWithLinks';
 import { LinkInput, Row } from '../../../form';
 import { SimplePanel } from '../../panel';
-import { ExpandedTextInput } from '../../shared';
+import { Input } from '../../../../../design-system';
 
 const Error = styled.span`
```
Comment (samwhale) [inline assets/src/edit-story/components/panels/design/pageAttachment/pageAttachment.js]: I think we should update this to use a `<Text />` component. Then we won't have to use old colors and a font size

**Review by maxyinger — APPROVED** [review]

File: assets/src/edit-story/components/panels/design/pageAttachment/pageAttachment.js [hunk 4]
```diff
@@ -108,16 +107,39 @@ function PageAttachmentPanel() {
...
+    [_ctaText, cancelCTAUpdate, defaultCTA, updatePageAttachment]
+  );
+
   return (
     <SimplePanel
       name="pageAttachment"
       title={__('Page Attachment', 'web-stories')}
     >
       <LinkInput
-        description={__(
```
Comment (BrittanyIRL) [inline assets/src/edit-story/components/panels/design/pageAttachment/pageAttachment.js:117]: Is this description unnecessary now?

File: assets/src/edit-story/components/panels/design/pageAttachment/pageAttachment.js [hunk 5]
```diff
@@ -108,16 +107,39 @@ function PageAttachmentPanel() {
...
+    ({ target }) => {
+      const { value } = target;
+      // This allows smooth input value change without any lag.
+      _setCtaText(value);
+      debouncedCTAUpdate(value);
+    },
+    [debouncedCTAUpdate]
+  );
+
+  const handleBlur = useCallback(
```
Comment (BrittanyIRL) [inline assets/src/edit-story/components/panels/design/pageAttachment/pageAttachment.js:117]: I think we should check for trim() on blur to block people like me trying to break it with 5 spaces?

Kapture 2021-03-04 at 14 37 24

Kapture 2021-03-04 at 14 38 50

**Review by BrittanyIRL — APPROVED** [review]
looks great! one nit!

File: assets/src/edit-story/components/panels/design/pageAttachment/pageAttachment.js (hunk 4 above)
Comment (embarks) [reply assets/src/edit-story/components/panels/design/pageAttachment/pageAttachment.js:117]: It's not visible anymore in the figma, maybe this should be a label somewhere though

File: assets/src/edit-story/components/panels/design/pageAttachment/pageAttachment.js [hunk 6]
```diff
@@ -108,16 +107,39 @@ function PageAttachmentPanel() {
...
       title={__('Page Attachment', 'web-stories')}
     >
       <LinkInput
-        description={__(
-          'Type an address to add a page attachment',
-          'web-stories'
-        )}
         onChange={(value) => updatePageAttachment({ url: value })}
         onFocus={onFocus}
         value={url || ''}
```
Comment (barklund) [inline assets/src/edit-story/components/panels/design/pageAttachment/pageAttachment.js]: There's something funky going on around here when multiple elements are selected with different link targets. When you select the elements, the input correctly says "Mixed", but if you focus and blur the input without entering anything, the input switches to `"http://((MULTIPLE))"` which I'm pretty sure is wrong.

Screen Shot 2021-03-04 at 23 33 12

However, I'm inclined to say that we just merge this as-is and we fix all the Mixed stuff in a single follow-up PR "later" (still in this release), as we're doing it in many different ways in different PR's.

**Review by barklund — APPROVED** [review]
In general this looks good and works as it's supposed to. There is an edge-case issue with Mixed, but I think it's better to merge and fix all Mixed-issues at "the end" before release.

File: assets/src/edit-story/components/panels/design/pageAttachment/pageAttachment.js (hunk 6 above)
Comment (miina) [reply assets/src/edit-story/components/panels/design/pageAttachment/pageAttachment.js]: The multi-value handling was not implemented to the link input indeed... For the Page Attachment panel it actually doesn't matter since only one Page can be selected at a time.

File: assets/src/edit-story/components/panels/design/pageAttachment/pageAttachment.js [hunk 7]
```diff
@@ -127,7 +149,7 @@ function PageAttachmentPanel() {
 
       {displayWarning && (
         <Row>
-          <Error>
+          <Error forwardedAs="span" size={PRESET_SIZES.X_SMALL}>
```
Comment (miina) [inline assets/src/edit-story/components/panels/design/pageAttachment/pageAttachment.js]: I wonder if this should use the input's Errored version instead with the warning message displayed as the `Hint`. This warning is not in the design currently (I think, couldn't find it at least), so not sure if there are changes needed. Might be good to confirm with Sam (can be followed up in a separate PR though if any changes are needed)

Comment (embarks) [reply assets/src/edit-story/components/panels/design/pageAttachment/pageAttachment.js]: I retained the existing functionality for regression.

Comment (miina) [reply assets/src/edit-story/components/panels/design/pageAttachment/pageAttachment.js]: Makes sense, thought of the style mainly but it's probably good as is! 👍

**Comment by embarks** [comment]
That makes sense. Easy add too, I'll include it. :)

File: assets/src/edit-story/components/panels/design/pageAttachment/pageAttachment.js (hunk 7 above)
Comment (embarks) [reply assets/src/edit-story/components/panels/design/pageAttachment/pageAttachment.js]: That makes sense. Easy add too, I'll include it. :)

File: assets/src/edit-story/components/panels/design/pageAttachment/pageAttachment.js (hunk 4 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/panels/design/pageAttachment/pageAttachment.js:117]: probs just for a11y
